### PR TITLE
feat(escalating-issues): enforce substatus values

### PR DESCRIPTION
**Pull request by barkbarkimashark**
After we backfill and default legal values to substatus, we want to enforce it before we save.

### CODE REVIEW

**Comment by jangjodi** [comment]
just curious - in what scenarios would the illegal combinations of 
1. `status=GroupStatus.UNRESOLVED` and `substatus!=GroupSubStatus.ONGOING || GroupSubStatus.ESCALATING` and
2. `status=GroupStatus.IGNORED` and `substatus!=GroupSubStatus.UNTIL_ESCALATING` 
occur?

File: src/sentry/models/group.py [hunk 1]
```diff
@@ -716,3 +716,18 @@ def pre_save_group_default_substatus(instance, sender, *args, **kwargs):
             instance.substatus = GroupSubStatus.ONGOING
         if instance.status == GroupStatus.IGNORED and instance.substatus is None:
             instance.substatus = GroupSubStatus.UNTIL_ESCALATING
+
+        if instance.status == GroupStatus.UNRESOLVED and instance.substatus not in (
+            GroupSubStatus.ONGOING,
```
Comment (snigdhas) [inline src/sentry/models/group.py:721]: I believe we also want to add GroupSubStatus.Regressed

Comment (barkbarkimashark) [reply src/sentry/models/group.py:721]: Do we also want to add `NEW` as well?

Comment (barkbarkimashark) [reply src/sentry/models/group.py:721]: Actually, I think I'll wait to add those in a new PR. Will also refactor the code a bit to codify the legal substatus mappings to the Group.status.

**Comment by barkbarkimashark** [comment]
> just curious - in what scenarios would the illegal combinations of
> 
>     1. `status=GroupStatus.UNRESOLVED` and `substatus!=GroupSubStatus.ONGOING || GroupSubStatus.ESCALATING` and
> 
>     2. `status=GroupStatus.IGNORED` and `substatus!=GroupSubStatus.UNTIL_ESCALATING`
>        occur?

Most likely programmer error. Ideally, everyone making changes to a group's substatus would have tests to verify all possible state transitions. With this change, at least we avoid writing to the DB with invalid states.

**Comment by barkbarkimashark** [comment]
Closing this since Ignored should default to null. Will re-open with modifications to handle Unresolved.

**Pull request closed without merging**